rpm/build/SPECS/jenkins.spec

```diff
@@ -45,103 +42,42 @@ rm -rf "%{buildroot}"
 %__install -d "%{buildroot}%{workdir}"
 %__install -d "%{buildroot}%{workdir}/plugins"
 
-%__install -d "%{buildroot}/var/log/%{name}"
-%__install -d "%{buildroot}/var/cache/%{name}"
+%__install -d "%{buildroot}%{_localstatedir}/log/%{name}"
+%__install -d "%{buildroot}%{_localstatedir}/cache/%{name}"
 
-%__install -D -m0755 "%{SOURCE1}" "%{buildroot}/etc/init.d/%{name}"
-%__sed -i 's,~~WAR~~,%{_javadir}/%{name}.war,g' "%{buildroot}/etc/init.d/%{name}"
-%__install -d "%{buildroot}/usr/sbin"
-%__ln_s "../../etc/init.d/%{name}" "%{buildroot}/usr/sbin/rc%{name}"
-
-%__install -D -m0600 "%{SOURCE2}" "%{buildroot}/etc/sysconfig/%{name}"
-%__sed -i 's,~~HOME~~,%{workdir},g' "%{buildroot}/etc/sysconfig/%{name}"
-
-%__install -D -m0644 "%{SOURCE3}" "%{buildroot}/etc/logrotate.d/%{name}"
+%__install -D -m0644 "%{SOURCE3}" "%{buildroot}%{_sysconfdir}/logrotate.d/%{name}"
 
 %__install -D -m0644 "%{SOURCE4}" "%{buildroot}%{_unitdir}/%{name}.service"
+
 %__install -D -m0755 "%{SOURCE5}" "%{buildroot}%{_bindir}/%{name}"
-%__install -d "%{buildroot}%{_datadir}/%{name}"
-%__install -D -m0755 "%{SOURCE6}" "%{buildroot}%{_datadir}/%{name}/migrate"
 
 %pre
 /usr/bin/getent group %{name} &>/dev/null || /usr/sbin/groupadd -r %{name} &>/dev/null
 # SUSE version had -o here, but in Fedora -o isn't allowed without -u
 /usr/bin/getent passwd %{name} &>/dev/null || /usr/sbin/useradd -g %{name} -s /bin/false -r -c "@@SUMMARY@@" \
 	-d "%{workdir}" %{name} &>/dev/null
 
-  # Used to decide later if we should perform a chown in case JENKINS_INSTALL_SKIP_CHOWN is false
-  # Check if a previous installation exists, if so check the JENKINS_HOME value and existing owners of work, log and cache dir, need to to this check
-  # here because the %files directive overwrites folder owners, I have not found a simple way to make the
-  # files directive to use JENKINS_USER as owner.
-  if [ -f "/etc/sysconfig/%{name}" ]; then
-      logger -t %{name}.installer "Found previous config file /etc/sysconfig/%{name}"
-      . "/etc/sysconfig/%{name}"
-      stat --format=%U "/var/cache/%{name}" > "/tmp/%{name}.installer.cacheowner"
-      stat --format=%U "/var/log/%{name}"  >  "/tmp/%{name}.installer.logowner"
-      stat --format=%U ${JENKINS_HOME:-%{workdir}}  > "/tmp/%{name}.installer.workdirowner"
-  else
-      logger -t %{name}.installer "No previous config file /etc/sysconfig/%{name} found"
-  fi
-
 %post
-%{_datadir}/%{name}/migrate "/etc/sysconfig/%{name}" || true
```

Comment:
We shipped support for systemd over a year ago, so I think it is safe to drop support for migrating from System V configuration. :+1:

Reply:
Is there any risk that users of a recent release (e.g., the last LTS) might have been still using the System V init configuration rather than the systemd configuration by accident, simply by running the `/etc/init.d` scripts rather than using the `service` or `systemctl` commands?

Perhaps a more conservative choice would be to change https://github.com/jenkinsci/packaging/blob/c3546bf96c4e7a48a0c55c3d3900ab76bea87a85/systemd/migrate.sh#L440 to simply exit 0 rather 1 when the System V configuration does not exist (or, equivalently, change the caller to avoid calling `migrate.sh` when the System V configuration does not exist). `migrate.sh` already exits 0 when `/etc/systemd/system/jenkins.service.d/override.conf` exists. If we make the change I suggested, then we could continue shipping the migration script, which would just do nothing for new installs or upgrades from a systemd-based installation, while continuing to support migration from an older System V configuration.

The reason I'm worried about this is that if this breaks some user's configuration, I'm going to get the support call. Feel free to let me know if you think I'm overthinking this or being overly paranoid.

Reply:
Why would anyone want to invoke `/etc/init.d/jenkins` directly? Best I could imagine is to have a poorly configured process monitor/supervisor like `monit`, `runit` or `supervisord`. These can easily be set up to use the systemd unit instead of the script. If we want `service` to use the systemd unit, we need to remove `/etc/init.d/jenkins` completely, which would break that hypothetical scenario (process supervisors) anyway.

Regarding the `migration` script – I see your point. How about that?
```diff
 %post
-%systemd_post %{name}.service
 if [ $1 -eq 1 ]; then
     %__install -d -m 0755 -o %{name} -g %{name} %{workdir}
     %__install -d -m 0750 -o %{name} -g %{name} %{_localstatedir}/cache/%{name}
+elif [ -f "%{_sysconfdir}/sysconfig/%{name}" ]; then
+    %{_datadir}/%{name}/migrate "/etc/sysconfig/%{name}" || true
 fi
+%systemd_post %{name}.service
```

Reply:
> Why would anyone want to invoke /etc/init.d/jenkins directly?

You are right, I cannot see a good reason for that.

> If we want service to use the systemd unit

Yes, I think we do want `service` to use the systemd unit.